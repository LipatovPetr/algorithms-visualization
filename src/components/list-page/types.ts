import { ElementStates } from "../../types/element-states";

export type linkedlistModelNode = {
  value: string;
  state: ElementStates;
  incomingValue: null | string;
};
