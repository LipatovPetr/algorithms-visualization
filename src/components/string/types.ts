import { ElementStates } from "../../types/element-states";

export type charObj = {
  character: string;
  state: ElementStates;
};

export type stringMappedToCharsWithState = Array<charObj>;
