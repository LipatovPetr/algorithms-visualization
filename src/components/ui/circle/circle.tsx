import React from "react";
import styles from "./circle.module.css";
import { ElementStates } from "../../../types/element-states";

interface CircleProps {
  state?: ElementStates;
  letter?: string;
  head?: string | React.ReactElement | null;
  index?: number;
  tail?: string | React.ReactElement | null;
  tailType?: "string" | "element";
  extraClass?: string;
  isSmall?: boolean;
}

export const Circle: React.FC<CircleProps> = ({
  state = ElementStates.Default,
  letter,
  head,
  index,
  tail,
  extraClass = "",
  isSmall,
}) => {
  return (
    <div
      className={`${styles.content} ${extraClass}`}
      data-testid="circle-component"
      data-test="circle-component"
    >
      <div
        data-test="head"
        className={`text text_type_input text_color_input mb-4 ${
          styles.absolute
        } ${styles.head} ${
          styles[typeof head === "string" ? "string" : "element"]
        }`}
      >
        {head}
      </div>
      <div
        data-testid="circle-element"
        data-test="circle-element"
        className={`${styles.circle}  ${isSmall ? styles.small : ""} ${
          styles[state]
        }`}
      >
        <p
          className={`text text_type_circle text_color_input ${styles.letter}`}
          data-test="circle-letter"
        >
          {letter}
        </p>
      </div>
      <p
        className={`text text_type_input text_color_input mt-4 ${styles.absolute} ${styles.index}`}
      >
        {index?.toString()}
      </p>
      <div
        data-test="tail"
        className={`text text_type_input text_color_input mt-4 ${
          styles.absolute
        } ${index?.toString() ? styles.tail60 : styles.tail30} ${
          styles[typeof tail === "string" ? "string" : "element"]
        }`}
      >
        {tail}
      </div>
    </div>
  );
};
