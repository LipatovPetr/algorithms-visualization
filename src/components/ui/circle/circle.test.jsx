import { render, screen, fireEvent } from "@testing-library/react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("Circle Component", () => {
  test("Render Circle without props", () => {
    render(<Circle />);
    const circle = screen.getByTestId("circle-component");
    expect(circle).toBeInTheDocument();
    expect(circle).toMatchSnapshot();
  });

  test("Render Circle with letters", () => {
    render(<Circle letter="qwe" />);
    const letters = screen.getByText("qwe");

    expect(letters).toBeInTheDocument();
    expect(letters).toMatchSnapshot();
  });

  test("Render Circle with head", () => {
    render(<Circle head="head" />);
    const head = screen.getByText("head");

    expect(head).toBeInTheDocument();
    expect(head).toMatchSnapshot();
  });

  test("Render Circle with tail", () => {
    render(<Circle tail="tail" />);
    const tail = screen.getByText("tail");

    expect(tail).toBeInTheDocument();
    expect(tail).toMatchSnapshot();
  });
  test("Render Circle with index", () => {
    render(<Circle index={12} />);
    const index = screen.getByText(12);

    expect(index).toBeInTheDocument();
    expect(index).toMatchSnapshot();
  });

  test("Render Circle with small class", () => {
    render(<Circle isSmall={true} />);
    const circleEl = screen.getByTestId("circle-element");

    expect(circleEl).toHaveClass("small");
    expect(circleEl).toMatchSnapshot();
  });

  test("Render Circle with DEFAULT state", () => {
    render(<Circle state={ElementStates.Default} />);
    const circleEl = screen.getByTestId("circle-element");

    expect(circleEl).toHaveClass("default");
    expect(circleEl).toMatchSnapshot();
  });

  test("Render Circle with CHANGING state", () => {
    render(<Circle state={ElementStates.Changing} />);
    const circleEl = screen.getByTestId("circle-element");

    expect(circleEl).toHaveClass("changing");
    expect(circleEl).toMatchSnapshot();
  });

  test("Render Circle with MODIFIED state", () => {
    render(<Circle state={ElementStates.Modified} />);
    const circleEl = screen.getByTestId("circle-element");

    expect(circleEl).toHaveClass("modified");
    expect(circleEl).toMatchSnapshot();
  });
});
