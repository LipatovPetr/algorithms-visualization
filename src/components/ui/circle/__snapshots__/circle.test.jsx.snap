// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle Component Render Circle with CHANGING state 1`] = `
<div
  class="circle   changing"
  data-test="circle-element"
  data-testid="circle-element"
>
  <p
    class="text text_type_circle text_color_input letter"
    data-test="circle-letter"
  />
</div>
`;

exports[`Circle Component Render Circle with DEFAULT state 1`] = `
<div
  class="circle   default"
  data-test="circle-element"
  data-testid="circle-element"
>
  <p
    class="text text_type_circle text_color_input letter"
    data-test="circle-letter"
  />
</div>
`;

exports[`Circle Component Render Circle with MODIFIED state 1`] = `
<div
  class="circle   modified"
  data-test="circle-element"
  data-testid="circle-element"
>
  <p
    class="text text_type_circle text_color_input letter"
    data-test="circle-letter"
  />
</div>
`;

exports[`Circle Component Render Circle with head 1`] = `
<div
  class="text text_type_input text_color_input mb-4 absolute head string"
  data-test="head"
>
  head
</div>
`;

exports[`Circle Component Render Circle with index 1`] = `
<p
  class="text text_type_input text_color_input mt-4 absolute index"
>
  12
</p>
`;

exports[`Circle Component Render Circle with letters 1`] = `
<p
  class="text text_type_circle text_color_input letter"
  data-test="circle-letter"
>
  qwe
</p>
`;

exports[`Circle Component Render Circle with small class 1`] = `
<div
  class="circle  small default"
  data-test="circle-element"
  data-testid="circle-element"
>
  <p
    class="text text_type_circle text_color_input letter"
    data-test="circle-letter"
  />
</div>
`;

exports[`Circle Component Render Circle with tail 1`] = `
<div
  class="text text_type_input text_color_input mt-4 absolute tail30 string"
  data-test="tail"
>
  tail
</div>
`;

exports[`Circle Component Render Circle without props 1`] = `
<div
  class="content "
  data-test="circle-component"
  data-testid="circle-component"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-test="head"
  />
  <div
    class="circle   default"
    data-test="circle-element"
    data-testid="circle-element"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-test="circle-letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-test="tail"
  />
</div>
`;
